spring-webflux/src/main/java/org/springframework/web/reactive/socket/adapter/JettyWebSocketSession.java

```diff
@@ -51,52 +71,77 @@ public JettyWebSocketSession(Session session, HandshakeInfo info, DataBufferFact
 	public JettyWebSocketSession(Session session, HandshakeInfo info, DataBufferFactory factory,
 			@Nullable Sinks.Empty<Void> completionSink) {
 
-		super(session, ObjectUtils.getIdentityHexString(session), info, factory, completionSink);
-		// TODO: suspend causes failures if invoked at this stage
-		// suspendReceiving();
+		super(session, ObjectUtils.getIdentityHexString(session), info, factory);
+		this.handlerCompletionSink = completionSink;
+		this.flux = Flux.create(emitter -> {
+			this.sink = emitter;
+			emitter.onRequest(n -> {
+				boolean demand = false;
+				this.lock.lock();
+				try {
+					this.requested += n;
+					if (!this.awaitingMessage && this.requested > 0) {
+						this.requested--;
```

Comment:
I'm not entirely sure how it translates in terms of `getDelegate().demand()` calls, but if the requested amount is unbounded, this decrement shouldn't happen.

with the above suggested change, unbounded request amount (`n == Integer.MAXVALUE` at any point) leads to `this.requested == Long.MAX_VALUE` so if that's the case the `this.requested--` decrement should be skipped.

Reply:
done.